src/client/datascience/interactive-common/notebookServerProvider.ts

```diff
@@ -81,7 +81,11 @@ export class NotebookServerProvider implements IJupyterServerProvider {
         const property = options.localJupyter ? 'local' : 'remote';
         if (!this.serverPromise[property]) {
             // Start a server
-            this.serverPromise[property] = this.startServer(options.resource, options.token, options.localJupyter);
+            this.serverPromise[property] = this.startServer(
+                options.resource,
+                options.tokenSource,
```

Comment:
Same as other places, this doesn't seem right, i.e passing the tokenSource

Reply:
See my other comment. We need a tokenSource now. Perhaps in some places we don't care, but I sided on the forcing it to be passed.